runtime/jcl/common/reflecthelp.c

```diff
@@ -1720,3 +1720,233 @@ getFieldsHelper(JNIEnv *env, jobject cls)
 	vmFuncs->internalExitVMToJNI(vmThread);
 	return result;
 }
+
+/* Determine whether currentMethod is the accessor method for a record component with name componentName.
+ * an accessor method will have the same name as the component and take zero parameters.
+ */
+static U_8
+isRecordComponentAccessorMethodMatch(J9Method *currentMethod, const char* componentName, const char* componentSignature) {
+	J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(currentMethod);
+	J9UTF8 *methodNameUtf8 = J9ROMMETHOD_NAME(romMethod);
+	U_16 methodLength = J9UTF8_LENGTH(methodNameUtf8);
+	const U_8 *methodName = J9UTF8_DATA(methodNameUtf8);
+	if ((methodLength == strlen(componentName)) 
+		&& (0 == strcmp((const char*)methodName, componentName))
+	) {
+		/* verify parameters. An accessor will not have any input parameters. */
+		J9UTF8 *methodSignatureUtf8 = J9ROMMETHOD_SIGNATURE(romMethod);
+		const U_8 *methodSignature = J9UTF8_DATA(methodSignatureUtf8);
+		U_16 methodSignatureLength = J9UTF8_LENGTH(methodSignatureUtf8);
+
+		if ((methodSignatureLength >= 2)
+			&& ('(' == methodSignature[0]) && (')' == methodSignature[1])
+			&& (((methodSignatureLength == 3) && componentSignature[2] == methodSignature[2])
+				|| (0 == strncmp((const char*)methodSignature + 2, componentSignature, methodSignatureLength - 2))
+			)
+		) {
+			return TRUE;
+		}
+	}
+	return FALSE;
+}
```

Comment:
We've seen when doing other name/sig searches that it is more efficient to check the lengths first before doing any strcmps.  Something like this (not tested) does the length checks up front and has them passed in as this is called in a loop:
```c
static U_8
isRecordComponentAccessorMethodMatch(J9Method *currentMethod, const char* componentName, const U_16 nameLength, const char* componentSignature, const U_16 sigLength) {
        J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(currentMethod);
	J9UTF8 *methodNameUtf8 = J9ROMMETHOD_NAME(romMethod);
	U_16 methodLength = J9UTF8_LENGTH(methodNameUtf8);
	const U_8 *methodName = J9UTF8_DATA(methodNameUtf8);

	/* verify parameters. An accessor will not have any input parameters. */
	J9UTF8 *methodSignatureUtf8 = J9ROMMETHOD_SIGNATURE(romMethod);
	const U_8 *methodSignature = J9UTF8_DATA(methodSignatureUtf8);
	U_16 methodSignatureLength = J9UTF8_LENGTH(methodSignatureUtf8);

	/* Compare name and signature including the `()` for the accessor */
	if ((methodLength == nameLength) 
	&& (methodSignatureLength == (sigLength + 2))
	) {
		if (0 == strcmp((const char*)methodName, componentName)) {
 			if (0 == strncmp((const char*)methodSignature + 2, componentSignature, methodSignatureLength - 2)) {
				return TRUE;
			}
		}
	}
	return FALSE;
}
```